### ISSUE TITLE
Failure to test RMWC form inputs using React Testing Library

### ISSUE THREAD

**Issue by majames** [jamesmfriedman/rmwc#504]
- What RMWC Version are you using [major.minor.patch]: **5.7.0**
- Name your build system [Webpack, Rollup...]: Jest and [React Testing Library](https://testing-library.com/docs/react-testing-library/intro)

It is currently impossible to test components which have multiple form inputs as children using [`getByLabelText()`](https://testing-library.com/docs/dom-testing-library/api-queries#bylabeltext) utility exposed by `react-testing-library`. 

This is because all form fields are given the [same `id`s in the test environment](https://github.com/jamesmfriedman/rmwc/blob/master/src/base/utils/random-id.tsx), and, therefore, attempting to trigger events on them (e.g.`click`) **always refers to the first input instead of the input it is actually labeling.**

I'm assuming `randomId()` behaviour was overwritten in a testing environment because internal RMWC tests depend on the hardcoded value? It may be worth updating the tests within RMWC to better reflect what users "see", but super interested on your thoughts for the best solution here!

**Comment by jamesmfriedman** [comment]
The hard coded value was put into place to support snapshot testing. Without it, the snapshot tests would fail due to always having a different ID. 

If you need to reference them by a specific ID you can pass one into the component itself.

**Comment by jamesmfriedman** [comment]
Thought about this more and one option is to use the label instead of the hard coded value in the test environment. This would cause issues in the edge case that you have two labels that were the same but it seems like an improvement to the current behavior. What do you think?

**Comment by majames** [comment]
That would be great! In our application we try to follow the guiding principles outlined in the [react testing library docs](https://testing-library.com/docs/guiding-principles) so we try to avoid referencing dom nodes by ID 🙂

**Comment by jamesmfriedman** [comment]
Oh for sure. It would fix this particular issue by hardcoding the ID to something else but I’ll look at the label fix.

**Comment by jamesmfriedman** [comment]
I'm pretty certain the following code will address the issue. Basically tries ID, a label, and then falls back to random.

```typescript
import { useState } from 'react';
import { randomId } from './random-id';

export const useId = (
  prefix: string,
  props: { [key: string]: any }
): string => {
  const idToUse = props.id
    ? props.id
    : props.label
    ? `${prefix}-${props.label}`
    : randomId(prefix);
  const [id] = useState(idToUse);
  return id;
};

```

**Comment by jamesmfriedman** [comment]
Fixed in 6.0.0.
